### PR TITLE
RIA-8796 - Set listing location and whether hearing is remote when system triggers "List the case"

### PR DESCRIPTION
**Pull request by robin-ieong**
…stem triggers List the case event

**Before creating a pull request make sure that:**

- [x ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-8796

### Change description ###
Set listing location and whether hearing is remote when system triggers "List the case"
Also rollback previous changes to keep the existing implementation when refdata is enabled

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/handlers/servicedatahandlers/ListCaseHandler.java [hunk 1]
```diff
@@ -53,8 +58,19 @@ public ServiceDataResponse<ServiceData> handle(ServiceData serviceData) {
...
+        log.info("asylumCase for  Case ID `{}` contains '{}'", caseId, asylumCase.toString());
 
-        updateListCaseHearingDetails(serviceData, asylumCase);
+        boolean isAppealsLocationRefDataEnabled = false;
+        try {
+            isAppealsLocationRefDataEnabled = featureToggler.getValueAsServiceUser(
+                APPEALS_LOCATION_REF_DATA_FEATURE, false);
+        } catch (Exception ex) {
+            log.info("Error occurred during isBailsLocationRefDataEnabled : " + ex);
+        }
```
Comment (augustio) [inline src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/handlers/servicedatahandlers/ListCaseHandler.java:69]: Not sure why you have this try/catch here. Do we expect FeatureToggler::getValueAsServiceUser to throw exception?
In the log message of the catch block, I suppose you mean 'isAppealsLocationRefDataEnabled'?

**Review by augustio — CHANGES_REQUESTED** [review]
Left a comment.

File: src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/handlers/servicedatahandlers/ListCaseHandler.java (hunk 1 above)
Comment (robin-ieong) [reply src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/handlers/servicedatahandlers/ListCaseHandler.java]: It was used for debug purposes previously, but I have removed it now as there should be minimal risks of it throwing error after the issue has been addressed

**Review by augustio — APPROVED** [review]
LGTM.

**Pull request merged**